### DISCUSSION TITLE
Install-Module dbatools    not working

### DISCUSSION THREAD

**Discussion by faizqur** [dataplat/dbatools#7450] [Q&A] [answered]
I am getting an error as shown below when I use Install-Module dbatools with Powershell on Windows 10 having SQL server 2017 installed. How to resolve this issue, please advise .

Thanks
IQ

WARNING: Unable to download from URI 'https://go.microsoft.com/fwlink/?LinkID=627338&clcid=0x
409' to ''.
WARNING: Unable to download the list of available providers. Check your internet connection.
PackageManagement\Install-PackageProvider : No match was found for the specified search 
criteria for the provider 'NuGet'. The package provider requires 'PackageManagement' and 
'Provider' tags. Please check if the specified package has the tags.
At C:\Program Files\WindowsPowerShell\Modules\PowerShellGet\1.0.0.1\PSModule.psm1:7468 
char:21
+ ...     $null = PackageManagement\Install-PackageProvider -Name $script:N ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (Microsoft.Power...PackageProvider:InstallPa 
   ckageProvider) [Install-PackageProvider], Exception
    + FullyQualifiedErrorId : NoMatchFoundForProvider,Microsoft.PowerShell.PackageManagemen 
   t.Cmdlets.InstallPackageProvider
 
PackageManagement\Import-PackageProvider : No match was found for the specified search 
criteria and provider name 'NuGet'. Try 'Get-PackageProvider -ListAvailable' to see if the 
provider exists on the system.
At C:\Program Files\WindowsPowerShell\Modules\PowerShellGet\1.0.0.1\PSModule.psm1:7474 
char:21
+ ...     $null = PackageManagement\Import-PackageProvider -Name $script:Nu ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidData: (NuGet:String) [Import-PackageProvider], Excepti 
   on
    + FullyQualifiedErrorId : NoMatchFoundForCriteria,Microsoft.PowerShell.PackageManagemen 
   t.Cmdlets.ImportPackageProvider
 
WARNING: Unable to download from URI 'https://go.microsoft.com/fwlink/?LinkID=627338&clcid=0x
409' to ''.
WARNING: Unable to download the list of available providers. Check your internet connection.
PackageManagement\Get-PackageProvider : Unable to find package provider 'NuGet'. It may not 
be imported yet. Try 'Get-PackageProvider -ListAvailable'.
At C:\Program Files\WindowsPowerShell\Modules\PowerShellGet\1.0.0.1\PSModule.psm1:7478 
char:30
+ ... tProvider = PackageManagement\Get-PackageProvider -Name $script:NuGet ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (Microsoft.Power...PackageProvider:GetPackage 
   Provider) [Get-PackageProvider], Exception
    + FullyQualifiedErrorId : UnknownProviderFromActivatedList,Microsoft.PowerShell.Package 
   Management.Cmdlets.GetPackageProvider
 
Install-Module : NuGet provider is required to interact with NuGet-based repositories. 
Please ensure that '2.8.5.201' or newer version of NuGet provider is installed.
At line:1 char:1
+ Install-Module dbatools
+ ~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [Install-Module], InvalidOperationExcep 
   tion
    + FullyQualifiedErrorId : CouldNotInstallNuGetProvider,Install-Module

**Comment by potatoqualitee** [comment]
hey @faizqur - this generally happens because of a .NET issue

SSL settings may need to be changed: `[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12`

You may also potentially have an old version of PowerShellGet/PackageManagement (Save-Module PowerShellGet, PackageManagement to temp place then delete and replace).

**Reply by faizqur** [reply]
Tried this suggestion but continue to get the same error. Please help, how do I proceed further ?

**Comment by faizqur** [comment]
Any suggestions on how to resolve this issue ?

**Comment by luisflavio** [comment, marked as answer]
Hi @faizqur 
I had the same issue and I solved running these steps below

Open Powershell (as Admin)

Run the command
[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12

Run the command
Install-PackageProvider -Name NuGet

Run the command
Install-Module -Name DBATools -force

**Reply by potatoqualitee** [reply]
oh that is an easy solution, thank you, @luisflavio !
